Option to keep elements hidden, options page

duart38 opened the pull request:
## uses a combination of sync(for ekill settings) and storage.local (for removed elemts).

## things to keep in mind

1.this new feature is disabled by default. right click and go to the new options page to enable it and then click save

**option page also has an area to remove saved pages**

rhardih commented on README.md:
```diff
@@ -24,6 +28,19 @@ MIT: http://rhardih.mit-license.org
+- Added an options page 
+- made it possible to store elements a user has removed
+- added permission to access chrome.storage
+
```
I think we'll just keep everything under a new 1.5 release, but I'll just fix that in the squash/merge process.

duart38 commented on ekill.js:
```diff
@@ -1,37 +1,125 @@
+          elementDump[elem].remove(); // remove the element.
+          break; // break out of the loop if we found the element to keep that CPU nice and cold..
+        }
+      }
```
do you have any ideas here? i was thinking about the id attribute but i know allot of front-end devs that barely use id's  (kind of guilty of this myself) sometimes even ignoring classes all together and adding for example an empty span with some text in it.

rhardih replied:
I think it needs to be a mix of different things in order to provide a reliable solution, but I'll have to play around with some different approaches. This is a good enough start. 👍

duart38 commented:
## ok so i:
1. changed some code to make it work with firefox. 
2. also changed the settings page to be a bit nicer.
3. tried cleaning up the code
4. added a delayed function to handle elements that have not yet been added to the DOM

maticrivo commented:
Just a thought, haven't looked at your code yet..
Maybe instead of doing a delayed function you can take advantage of `MutationObserver` and observe the DOM to see when the element you want to kill is added to the DOM

rhardih commented:
Cool @duart38 👍 

I'll merge and do some experimentation on the grudge branch and ping you in the PR from grudge -> master.
